### ISSUE TITLE
Documentation - Configuration Handling

### ISSUE THREAD

**Issue by DancingPete** [pallets/flask#3589]
### Expected Behavior

While pouring through the documentation to learn the approaches to configuration handling, I spend time trying to replicate the subclass approach (thinking it was elegant):

```
class Config(object):
    DEBUG = False
    TESTING = False
    DATABASE_URI = 'sqlite:///:memory:'

class ProductionConfig(Config):
    DATABASE_URI = 'mysql://user@localhost/foo'

class DevelopmentConfig(Config):
    DEBUG = True

class TestingConfig(Config):
    TESTING = True
```
While it was possible to have the server run in DEBUG mode when I set the environment variable `FLASK_ENV` to development, or set the `DEBUG=1`, as well as when I ran a simple python module that set `app.config['DEBUG']=True` and ran the script with the `python app.py` command (where the name of the simple app script was app.py), it would not run in debug when I simply ran the flask server without explicitly setting the environment variable.

After a day of hunting I was finally able get clarity asking a question in Discord. It seems that the documentation refers back to configuration examples that preceded the flask CLI. 

I just wanted to share this experience and suggest removing the DEBUG configuration setting from the config file documentation examples to help newbies like myself avoid a day of hair loss. 

I am very much enjoying learning the framework and all the documentation otherwise. Thanks.

```python
# Paste a minimal example that causes the problem.
```

### Actual Behavior

```pytb
Paste the full traceback if there was an exception.
```

### Environment

* Python version: 3.7.5
* Flask version: 1.1.2
* Werkzeug version: 1.0.1

**Comment by carc1n0gen** [comment]
I cannot reproduce this.  

```py
from flask import Flask

app = Flask(__name__)
app.config['DEBUG'] = True

app.run()
```

`python app.py`

```
$ python app.py
 * Serving Flask app "app" (lazy loading)
 * Environment: development
 * Debug mode: on
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
 * Restarting with stat
 * Debugger is active!
 * Debugger PIN: 106-266-633
```

Debug mode is on

**Comment by DancingPete** [comment]
Perhaps I misrepresented myself in my description of the issue. Your additional information is what I learned in my conversation on Discord and what I was suggesting would be a helpful addition to the existing documentation. Nowhere in the documentation (that I was able to find) does it say as clearly and explicit as you just have in your "additional information":

>[If] you use the flask cli, you must use environment variables, as FLASK_ENV will override the values of DEBUG, and it defaults to 'production' which sets DEBUG to false.

Thanks for taking the time to respond.
